generic/utils/ras.py.data/common_func.sh

```diff
@@ -0,0 +1,211 @@
+    exec_log "/etc/init.d/rtas_errd start"
+}
+
+################################################################################
+# clean_servicelog ()
+# Purpose: Truncate the RTAS events database and cleans
+# /var/log/platform and /var/spool/mail/root
+# Parameters: None
+# Returns: The exit code of the servicelog command
+################################################################################
+clean_servicelog ()
+{
+  exec_log "servicelog_manage --truncate events --force"
+  SLOG_EXIT=$?
+  rm -f /var/log/platform
+  rm -f /var/spool/mail/root
+  return $SLOG_EXIT
+}
+
+
+################################################################################
+# print_log_header ()
+# Purpose: Initializes the execution log truncating it (tee without -a flag)
+# Parameters: None
+# Returns: 0
+################################################################################
+print_log_header ()
+{
+cat << TEXT | tee $EXECUTION_LOG
+$(date +'%d/%m/%y %H:%M')
+********************************************************************************
+* RAS Testcase
+* $TEST_GROUP
+* $TEST_SCENARIO
+********************************************************************************
+
+TEXT
+
+return 0
+}
+
+################################################################################
+# print_log_footer ()
+# Purpose: Checks whether the testcase ended successfuly and prints its status
+# Parameters: None
+# Returns: 0
+################################################################################
+print_log_footer ()
+{ if [ $IS_FAILED = 0 ]; then
+      TEST_STATUS='PASS'
+  else
+      TEST_STATUS='FAILED'
+      print_log "Showing testcase errors:"
+      exec_log "cat $ERROR_LOG"
+  fi
+
+cat << TEXT | tee -a $EXECUTION_LOG
+
+********************************************************************************
+* RAS Testcase
+* $TEST_GROUP
+* $TEST_SCENARIO
+* Test status: $TEST_STATUS
+********************************************************************************
+$(date +'%d/%m/%y %H:%M')
+TEXT
+
+  return 0
+}
+
+check_firmware ()
+{
+  FIRMWARE_SRC="http://w3.austin.ibm.com/afs/austin/projects/gfw/images/Squadrons/latest/"
+  LATEST_IMAGE=01SF240_291_201.img
+  FIRMWARE_DIR=/root/firmware-images
+  mkdir -p $FIRMWARE_DIR
+
+  # If the latest firmware file doesn't exist, download it
+  if [ ! -e $FIRMWARE_DIR/$LATEST_IMAGE ]; then
+      print_log "We didn't find the latest image available, downloading it..."
+      cd $FIRMWARE_DIR
+      wget $FIRMWARE_SRC/$LATEST_IMAGE
+  fi
+}
+
+################################################################################
+# manual_test_dialog ()
+# Purpose: Dialog that will need the user's input regarding the behavior of an
+# interactive application (manual test)
+# Parameters: String with the name of the manual test
+# Returns: 0 if the manual test went OK, 1 otherwise
+################################################################################
+manual_test_dialog ()
```

Comment:
Oh, this could be a problem as avocado would wait for input for the infinity. See a recommendation about this in the cmd....